wgpu-core/src/track/buffer.rs

```diff
@@ -304,17 +304,17 @@ impl<A: HalApi> ResourceTracker for BufferTracker<A> {
                 //so it's already been released from user and so it's not inside Registry\Storage
                 if existing_ref_count <= 2 {
                     resource_log!(
-                        "BufferTracker::remove_abandoned: removing {:?}",
-                        self.metadata.get_resource_unchecked(index).as_info().id()
+                        "BufferTracker::remove_abandoned: removing {}",
+                        self.metadata.get_resource_unchecked(index).error_ident()
```

Comment:
> Wouldn't it be good to still print an ID, if we have one?

Once we remove the `id` field in `ResourceInfo` we will no longer have one. If we deem it necessary we can always introduce back a mechanism to add more data to `wgpu-core` objects that then gets printed in errors/logs.

> issue(non-blocking): Docs. for `Resource::label` [state](https://github.com/ErichDonGubler/wgpu/blob/64a131feae3cc37fcfe003583e9a52be657abddd/wgpu-core/src/resource.rs#L182) that this will be non-empty unless provided a non-empty label from the user. However, `None` _can_ be specified as a `label`, in which case `ResourceInfo::new`  [appears to use an empty string](https://github.com/ErichDonGubler/wgpu/blob/64a131feae3cc37fcfe003583e9a52be657abddd/wgpu-core/src/resource.rs#L103). Between the docs., behavior, and my understanding, do you know which is incorrect?

`None` and empty `String`/`str` are currently equivalent in terms of behavior. On the spec side of things, the label will be set to an empty string if it wasn't specified. In our rust API (`wgpu`) the label type is `Option<&'a str>`. I don't know why it's not just a `&str`. Maybe that's another thing worth changing.